interactors/src/main/java/fr/coppernic/lib/interactors/ocr/OcrInteractorIdPlatform.java

```diff
@@ -0,0 +1,147 @@
+package fr.coppernic.lib.interactors.ocr;
+
+import android.content.Context;
+import android.os.SystemClock;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.inject.Inject;
+
+import fr.coppernic.external.elyctismrz.MrzScanner;
+import fr.coppernic.sdk.power.PowerManager;
+import fr.coppernic.sdk.power.api.PowerListener;
+import fr.coppernic.sdk.power.api.peripheral.Peripheral;
+import fr.coppernic.sdk.power.impl.idplatform.IdPlatformPeripheral;
+import fr.coppernic.sdk.utils.core.CpcResult;
+import io.reactivex.Single;
+import io.reactivex.SingleEmitter;
+import io.reactivex.SingleOnSubscribe;
+import timber.log.Timber;
+
+public class OcrInteractorIdPlatform implements OcrInteractor, PowerListener, MrzScanner.Listener {
+    private Context context;
+
+    // MRZ reader
+    // Elyctis ID Platform
+    private MrzScanner mrzReader;
+    private SingleEmitter<String> emitter;
+    private SingleEmitter<Boolean> powerEmitter;
+    private AtomicBoolean isReading = new AtomicBoolean(false);
+
+    @Inject
+    OcrInteractorIdPlatform(Context context) {
+        this.context = context;
+    }
+
+    @Override
+    public Single<Boolean> power(final Boolean power) {
```

Comment:
Ici tu peux récupérer le Single utilisé en interne dans l'API power au lieu d'en créer un nouveau. 

    return IdPlatformPeripheral.OCR.getDescriptor().power(context, power)

Que tu peux mapper selon tes besoins

    .map { 
       Timber.d("onPowerUp");
        if(power) {
            while (!mrzReader.open()) {
                Timber.d("Failed to open");
                SystemClock.sleep(100);
            }
        }
        return power
    }